src/Halogen/HTML/Events/Types.purs

```diff
@@ -33,6 +36,23 @@ type MouseEvent =
   , which :: Number
   )
 
+-- | Identifies the additional fields which are available on drag events.
+type DragEvent =
+  ( button :: Number
+  , detail :: Number
+  , relatedTarget :: HTMLElement
+  , clientX :: Number
+  , clientY :: Number
+  , screenX	:: Number
+  , screenY	:: Number
+  , ctrlKey	:: Boolean
+  , shiftKey :: Boolean
+  , altKey :: Boolean
+  , metaKey	:: Boolean
+  , which :: Number
+  , dataTransfer :: Nullable DataTransfer
```

Comment:
Yes, that's the thing that I don't like. With other event handlers you can `unsafeCoerce` it to some other record if the halogen's type doesn't mention a field that you need. A nice property when you need a temporary working solution. But constructing the records fairly (providing a minimal value which would satisfy the type) would break that implementation detail.

I am not worried about not having the possibility for hack (it's only better to provide some `self` property with original event rather than coercing the thing), but I would hate inconsistency, so if we do that for `drop` events I think we should do something like that for the rest of them (e.g. make them into newtypes with foreign instances).